src/status_im/chat/views/message_input.cljs

```diff
@@ -72,24 +80,57 @@
-                               :disable?          disable?
-                               :sendind-disabled? sending-disabled?})
-       (plain-input-options {:set-layout-size-fn set-layout-size
-                             :disable?           (or disable? (not active?))}))
-     {:placeholder-text-color :#c0c5c9
-      :auto-focus             (when command?
-                                (not (:fullscreen command)))
-      :accessibility-label    id/chat-message-input
-      :default-value          (if command?
-                                (or input-command "")
-                                (or input-message ""))}
-     (when command?
-       (get-options parameter type)))])
+   just-set? [:get :just-set-command?]
+   sending-disabled? [:chat-ui-props :sending-disabled?]
+   state {:input          (atom nil)
+          :focused?       (atom nil)
+          :3-symbols-flag (r/atom false)}]
+  {:component-did-update (fn []
+                           (when just-set?
+                             (dispatch [:set :just-set-command? false]))
+                           (when @(:3-symbols-flag state)
+                             (reset! (:3-symbols-flag state) false))
+                           (when (and command?
+                                      (not (:fullscreen command)))
+                             (.focus @(:input state))))}
+
+  (do
+    @(:3-symbols-flag state)
+    [text-input
+     (merge
+       (if command?
+         (command-input-options {:icon-width        icon-width
+                                 :disable?          disable?
+                                 :sendind-disabled? sending-disabled?
+                                 :flag              (:3-symbols-flag state)})
+         (plain-input-options {:set-layout-size-fn set-layout-size
+                               :disable?           (or disable? (not active?))}))
+       {:placeholder-text-color :#c0c5c9
+        :onFocus                #(reset! (:focused? state) true)
+        :onBlur                 #(reset! (:focused? state) false)
+        :auto-focus             (when command? (not (:fullscreen command)))
+        :ref                    #(reset! (:input state) %)
+        :accessibility-label    id/chat-message-input
+        ;; for some reason app crashes when this property is not nil and
+        ;; input is not focused
+        :selection              (cond (and command?
+                                           (= input-command command-prefix)
+                                           @(:focused? state)
+                                           (not just-set?))
+                                      {:start 2
+                                       :end   2}
+
+                                      (and @(:focused? state)
+                                           @(:3-symbols-flag state))
+                                      {:start 3
+                                       :end   3}
+
+                                      :else nil)
```

Comment:
as for me, it is better to move the code from `:selection` to a separate function for better readability